__tests__/app.test.js

```diff
@@ -54,32 +54,68 @@ describe('/api/articles/:article_id', () => {
         return request(app)
       .get('/api/articles/1')
       .expect(200)
-      .then((response) => {
-        expect(response.body.article.title).toBe("Living in the shadow of a great man");
-        expect(response.body.article.topic).toBe("mitch");
-        expect(response.body.article.author).toBe( "butter_bridge");
-        expect(response.body.article.body).toBe( "I find this existence challenging");
-        expect(response.body.article).toHaveProperty("created_at");
-        expect(response.body.article.votes).toBe(100);
-        expect(response.body.article.article_img_url).toBe("https://images.pexels.com/photos/158651/news-newsletter-newspaper-information-158651.jpeg?w=700&h=700");
+      .then(({body}) => {
+        expect(body.article.title).toBe("Living in the shadow of a great man");
+        expect(body.article.topic).toBe("mitch");
+        expect(body.article.author).toBe( "butter_bridge");
+        expect(body.article.body).toBe( "I find this existence challenging");
+        expect(body.article).toHaveProperty("created_at");
+        expect(body.article.votes).toBe(100);
+        expect(body.article.article_img_url).toBe("https://images.pexels.com/photos/158651/news-newsletter-newspaper-information-158651.jpeg?w=700&h=700");
       });
     });
     test('GET:404 sends an appropriate status and error message when given a valid but non-existent id', () => {
         return request(app)
           .get('/api/articles/999')
           .expect(404)
-          .then((response) => {
-            expect(response.body.msg).toBe('No article found for article_id: 999');
+          .then(({body}) => {
+            expect(body.msg).toBe('No article found for article_id: 999');
           });
       });
       test('GET:400 sends an appropriate status and error message when given an invalid id', () => {
         return request(app)
           .get('/api/articles/not-a-team')
           .expect(400)
-          .then((response) => {
-            expect(response.body.msg).toBe('Bad request');
+          .then(({body}) => {
+            expect(body.msg).toBe('Bad request');
           });
     });
 });
 
-// Consider what errors could occur with this endpoint, and make sure to test for them.
\ No newline at end of file
+describe('/api.articles', () => {
+    test("GET:200 responds with an object describing all available articles, ordered by most recent post", () => {
+        return request(app)
+          .get("/api/articles")
+          .expect(200)
+          .then(({ body }) => {
+            const { articles } = body;
+            expect(articles).toBeSortedBy('created_at', {
+                descending: true,
+                coerce: true
+              });
+              expect(articles.length).toBe(13);
+            for (const key in articles) {
```

Comment:
Articles is an array, so does not have a key. Your key here is the index of the element and you are checking the properties of that element 🙂